src/containers/flows/MakeABuy/MakeABuy.tsx

```diff
@@ -0,0 +1,494 @@
+	const { account, library } = walletContext;
+
+	const znsContracts = useZnsContracts()!;
+	const wildContract: ERC20 = znsContracts.wildToken;
+
+	const [isBuyValid, setBuyValid] = useState(false);
+
+	const sdk = useZnsSdk();
+	const domainId = domain.id;
+
+	const provider = library && new providers.Web3Provider(library.provider);
+	const signer = provider && provider.getSigner(account!);
+
+	///////////////
+	// Functions //
+	///////////////
+
+	const sdkBuy = async () => {
+		await (
+			await sdk.instance.getZAuctionInstanceForDomain(domainId)
+		)
+			.buyNow(
+				{
+					tokenId: domainId,
+					amount: buyNowPrice!.toString(),
+				},
+				signer!,
+			)
+			.then(async (transcation) => {
+				await transcation.wait().then(() => {
+					setShowProccesing(false);
+					setBuyNowFinishedSuccesfully(true);
+				});
+			})
+			.catch(() => {
+				setActionDeclined(true);
+			});
+	};
+
+	const buyNft = async () => {
+		setIsMetamaskWaiting(true);
+
+		if (typeof buyNowPrice !== 'undefined') {
+			await (
+				await sdk.instance.getZAuctionInstanceForDomain(domainId)
+			)
+				.getZAuctionSpendAllowance(await signer?.getAddress()!)
+				.then(async (allowedSpendAmount) => {
+					if (allowedSpendAmount.gt(buyNowPrice)) {
+						setIsMetamaskWaiting(false);
+						setShowProccesing(true);
+						sdkBuy();
+					} else {
+						await (await sdk.instance.getZAuctionInstanceForDomain(domainId))
+							.approveZAuctionSpendTradeTokens(signer!)
+							.then(() => {
+								setIsMetamaskWaiting(false);
+								setShowProccesing(true);
+								sdkBuy();
+							});
+					}
+				})
+				.catch(() => {
+					setActionDeclined(true);
+				});
+		}
+	};
+
+	/////////////
+	// Effects //
+	/////////////
+
+	useEffect(() => {
+		setBuyValid(
+			typeof buyNowPrice !== 'undefined' &&
+				typeof wildBalance !== 'undefined' &&
+				Number(ethers.utils.formatEther(buyNowPrice)) <= wildBalance &&
+				Number(ethers.utils.formatEther(buyNowPrice)) > 0,
+		);
+	}, [buyNowPrice, wildBalance]);
+
+	useEffect(() => {
+		if (!account) {
+			return;
+		}
+
+		setIsMetamaskWaiting(true);
+
+		const fetchTokenBalance = async () => {
+			const balance = await wildContract.balanceOf(account);
+			setWildBalance(parseInt(ethers.utils.formatEther(balance), 10));
+		};
+
+		fetchTokenBalance();
+
+		(async () => {
+			const isAbleToSpendTokens = await (
+				await sdk.instance.getZAuctionInstanceForDomain(domainId)
+			).getZAuctionSpendAllowance(await signer?.getAddress()!);
+			setIsMetamaskWaiting(false);
+
+			if (isAbleToSpendTokens) {
+				await (
+					await sdk.instance.getZAuctionInstanceForDomain(domainId)
+				)
+					.getBuyNowPrice(domainId, signer!)
+					.then((price) => {
+						setBuyNowPrice(price);
+					})
+					.catch(() => {
+						setActionDeclined(true);
+					});
+			} else {
+				await (
+					await sdk.instance.getZAuctionInstanceForDomain(domainId)
+				)
+					.approveZAuctionSpendTradeTokens(signer!)
```

Comment:
Not sure why this check is needed in multiple places

Reply:
I can't get nft buy price without this approval